### ISSUE TITLE
after update to 4.3.3 can't receive any push notification in DefaultBroadcastReceiver

### ISSUE THREAD

**Issue by mohammadidiab** [chat-sdk/chat-sdk-android#408]
after update to 4.3.3 can't receive any push notification in DefaultBroadcastReceiver

i do my custom DefaultBroadcastReceiver
   FirebasePushModule.activate();
  builder.setInboundPushHandlingEnabled(false);
  builder.setShowLocalNotifications(true);
 
` <receiver
            android:name=".push.BroadcastReceiver"
            android:enabled="true"
            android:exported="true"
            android:permission="com.google.android.c2dm.permission.SEND">
            <intent-filter>
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
                <category android:name="com.newline.multe" />
            </intent-filter>
        </receiver>`

     <meta-data
            android:name="com.google.firebase.messaging.default_notification_icon"
            android:resource="@drawable/ic_launcher"
            tools:replace="android:resource" />

        <meta-data
            android:name="com.google.firebase.messaging.default_notification_color"
            android:resource="@color/colorAccent"
            tools:replace="android:resource" />

**Comment by mohammadidiab** [comment]
@bensmiley help my please.

**Comment by bensmiley** [comment]
@mohammadidiab in the latest version we updated how push notifications are sent. You would need to follow [this guide](https://github.com/chat-sdk/chat-sdk-android#setup-firebase-cloud-functions) to reactivate them.

**Comment by dogacsenol** [comment]
@bensmiley is it possible to migrate to the latest version without breaking notifications between people using old versions?

**Comment by mohammadidiab** [comment]
@bensmiley 
@dogacsenol 

thanks 
but some time don't receive any notification
and in firebase console (functions) have an error

`Error: data must only contain string values
    at FirebaseMessagingError.Error (native)
    at FirebaseMessagingError.FirebaseError [as constructor] (/user_code/node_modules/firebase-admin/lib/utils/error.js:39:28)
    at FirebaseMessagingError.PrefixedFirebaseError [as constructor] (/user_code/node_modules/firebase-admin/lib/utils/error.js:85:28)
    at new FirebaseMessagingError (/user_code/node_modules/firebase-admin/lib/utils/error.js:241:16)
    at /user_code/node_modules/firebase-admin/lib/messaging/messaging.js:95:19
    at Array.forEach (native)
    at validateStringMap (/user_code/node_modules/firebase-admin/lib/messaging/messaging.js:93:22)
    at validateMessage (/user_code/node_modules/firebase-admin/lib/messaging/messaging.js:322:5)
    at Messaging.send (/user_code/node_modules/firebase-admin/lib/messaging/messaging.js:461:9)
    at getUserIds.then (/user_code/index.js:220:39)`

**Comment by ezet** [comment]
I'm having the same issue as @mohammadidiab above

**Comment by bensmiley** [comment]
I've double checked this and it's working on the default version. Can you tell me where you're seeing this error? Also which version of the Chat SDK are you using?

**Comment by ezet** [comment]
I'm using 4.3.3.

It seems to be caused by user.meta.name being undefined, which results in message.data.chat_sdk_push_title being set to undefined. 'name' is undefined for all android users in my db, only 'name-lowercase' is defined, set to an empty string. For iOS users, both 'name' and 'name-lowercase' is defined correctly.

Returning an empty string if meta.name is undefined might be a temporary workaround.

**Comment by ezet** [comment]
Both null and undefined values are invalid FCM values. getUserName should return a valid name or an empty string if one can't be found. At the moment it returns null if meta is null,  and undefined if meta.name is undefined.

`
function getUserName(usersRef, userId) {
    return usersRef.child(userId).child('meta').once('value').then((meta) => {
        let metaValue = meta.val();
        if (metaValue !== null && metaValue["name"] !== undefined) {
            return metaValue["name"];
        }
        return "";
    });
}
`

Though a missing name might suggest there are other issues?

**Comment by bensmiley** [comment]
@ezet Ok, I'll deploy two fixes the first will be that users will get a default name when they sign up. The second is that I'll add a null check in the push handler code.

**Comment by bensmiley** [comment]
Resolved in e6431b5 and `4.3.6`. The new push script fixes the bug in the case that a user doesn't have a name set. Also, going forward, users will be given a name on signup.

**Comment by muhammadsr** [comment]
I am still getting a similar issue

```
Error: data must only contain string values
    at FirebaseMessagingError.Error (native)
    at FirebaseMessagingError.FirebaseError [as constructor] (/user_code/node_modules/firebase-admin/lib/utils/error.js:39:28)
    at FirebaseMessagingError.PrefixedFirebaseError [as constructor] (/user_code/node_modules/firebase-admin/lib/utils/error.js:85:28)
    at new FirebaseMessagingError (/user_code/node_modules/firebase-admin/lib/utils/error.js:241:16)
    at /user_code/node_modules/firebase-admin/lib/messaging/messaging.js:95:19
    at Array.forEach (native)
    at validateStringMap (/user_code/node_modules/firebase-admin/lib/messaging/messaging.js:93:22)
    at validateMessage (/user_code/node_modules/firebase-admin/lib/messaging/messaging.js:322:5)
    at Messaging.send (/user_code/node_modules/firebase-admin/lib/messaging/messaging.js:461:9)
    at getUserIds.then (/user_code/index.js:224:39)
```
